### PR TITLE
fix tab control and tabs is not synced after focused tab is removed

### LINKED ISSUE #708: Incorrect tab behavior after using Ctrl+w
* Your Windows build number: 
  * Microsoft Windows [Version 10.0.18362.86]
  * Did a `git pull master` before using VS2019 to deploy locally.

* What you're doing and what's happening:
  1. To start, have 4 tabs open.
  2. Go to the 1st tab
  3. Hit Ctrl + w to close it. Terminal behaves as expected and closes the tab
  4. Both a and b below are where the bug shows up:
    a. Hit Ctrl + w again. The new 2nd tab (originally the 3rd tab) closes instead
    b. Hit Ctrl + Tab to cycle. Focus will change from the 1st tab to the 3rd tab (originally 4th) tab. 1st tab will also retain focused coloring.

  Also: If you start with 3 tabs open and then close the first one with Ctrl + w, you're not able to use Ctrl + Tab to cycle to the next tab.

  4a and 4b are not consecutive steps. They're different actions which can mess up after step 3

* What's wrong / what should be happening instead:
  For 4a: focused tab should close
  For 4b: 2nd tab should become selected, not the 3rd.

If there's any more information you need, please let me know!

### PR DESCRIPTION
**Pull request by bef178**
fix #708

### CODE REVIEW

File: src/cascadia/TerminalApp/App.cpp [hunk 1]
```diff
@@ -843,17 +823,26 @@ namespace winrt::TerminalApp::implementation
...
+        if (_tabs.size() <= 1) {
+            return;
+        }
+
         uint32_t tabIndexFromControl = 0;
         _tabView.Items().IndexOf(tabViewItem, tabIndexFromControl);
 
         if (tabIndexFromControl == _GetFocusedTabIndex())
         {
+            // switch to the "future" tab first, or current tab will not be controlled by tabview but remain focused and shown
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/App.cpp]: I don't think I understand what this comment is trying to say - could you clarify?

File: src/cascadia/TerminalApp/App.cpp [hunk 2]
```diff
@@ -843,17 +823,26 @@ namespace winrt::TerminalApp::implementation
     // - tabViewItem: the TabViewItem in the TabView that is being removed.
     void App::_RemoveTabViewItem(const IInspectable& tabViewItem)
     {
+        // TODO: GitHub:627: Need a better story for what should happen when the last tab is closed.
+        if (_tabs.size() <= 1) {
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/App.cpp]: nit: Our coding standard uses [Allman style](https://en.wikipedia.org/wiki/Indentation_style#Allman_style) braces, with braces on the line following the conditional.

**Comment by zadjii-msft** [comment]
@bef178 Could you make the title more descriptive? "Tab Sync" will be really hard to remember what it means when looking back at the commit log 3 months from now.

File: src/cascadia/TerminalApp/App.cpp (hunk 2 above)
Comment (bef178) [reply src/cascadia/TerminalApp/App.cpp]: Sorry about it. I will change.

File: src/cascadia/TerminalApp/App.cpp (hunk 1 above)
Comment (bef178) [reply src/cascadia/TerminalApp/App.cpp]: My poor expression... will remove this comment

**Pull request merged**